File: Ahorn/triggers/SecondTextboxTrigger.jl
module Sardine7SecondTextBoxTrigger

using ..Ahorn, Maple

@mapdef Trigger "Sardine7/SecondTextboxTrigger" SecondTextBoxTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, mode::String="OnPlayerEnter", dialog_id::String="", only_once::Bool=true, death_count::Integer=-1, lifespan::String="3.0")

const placements = Ahorn.PlacementDict(
    "Second Textbox (Sardine7)" => Ahorn.EntityPlacement(
        SecondTextBoxTrigger,
        "rectangle"
    )
)

function Ahorn.editingOptions(trigger::SecondTextBoxTrigger)
    return Dict{String, Any}(
        "mode" => Maple.mini_textbox_trigger_modes
    )
end

end

File: Loenn/triggers/second_textbox_trigger.lua
local enums = require("consts.celeste_enums")

local secondTextbox = {}

secondTextbox.name = "Sardine7/SecondTextboxTrigger"
secondTextbox.fieldInformation = {
    death_count = {
        fieldType = "integer",
    },
    lifespan = {
        fieldType = "number",
        minimumValue = 0.0
    },
    mode = {
        options = enums.mini_textbox_trigger_modes,
        editable = false
    }
}
secondTextbox.placements = {
    name = "second_textbox",
    data = {
        dialog_id = "",
        mode = "OnPlayerEnter",
        only_once = true,
        death_count = -1,
        lifespan = "3.0"
    }
}

return secondTextbox
